ports/raspberrypi/common-hal/busio/UART.c

```diff
@@ -78,7 +78,12 @@ static busio_uart_obj_t *active_uarts[NUM_UARTS];
 
 static void _copy_into_ringbuf(ringbuf_t *r, uart_inst_t *uart) {
     while (uart_is_readable(uart) && ringbuf_num_empty(r) > 0) {
-        ringbuf_put(r, (uint8_t)uart_get_hw(uart)->dr);
+        uint8_t ch = (uint8_t)uart_get_hw(uart)->dr;
+        if (ch == mp_interrupt_char) {
+            mp_sched_keyboard_interrupt();
+            continue;
+        }
+        ringbuf_put(r, ch);
```

Comment:
I don't understand why you are checking for ctrl-C in raw UART reading.

Reply:
Every port is meant to do this, but it was missing from this port. This is how ctrl-C is implemented. I discovered this because I was using the uart instead of usb cdc and ctrl-c didn't work.

Reply:
But people use UART's to send and receive binary data, not just to the REPL, and ctrl-C is a valid binary data character. I don't think you want to check at this level of the code.